External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -189,6 +192,45 @@ namespace FEXCore::Context {
     return NewThreadState;
   }
 
+  bool Context::HandleSegfault(FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
+    auto CTX = Thread->CTX;
+
+    auto FaultAddress = (uintptr_t)((siginfo_t*)info)->si_addr;
+
+    bool DoFlush = false;
+
+    {
+      FHU::ScopedSignalMaskWithSharedMutex lk(CTX->MemoryEntryMutex, true);
+
+      auto Entry = CTX->MemoryMaps.upper_bound(FaultAddress);
```

Comment:
Could you add a comment about the intended iteration behavior, similar to what you've written for the use of `lower_bound` below?

Reply:
Done